### PR TITLE
Feat :Add template flag to cyctl on module creation

### PR DESCRIPTION
**Pull request by shivamrazorpay**
closes #

## 📑 Description
- Added Template Flag to cycle on Module Creation.
- Fixes Issue : #439 

## ✅ Checks
- [ ] I have updated the documentation as required
- [x] I have performed a self-review of my code

### CODE REVIEW

File: cyctl/internal/create/modules.go [hunk 1]
```diff
@@ -74,7 +86,12 @@ var (
 		Args:    cobra.ExactArgs(1),
 		Aliases: []string{"modules"},
 		Run: func(cmd *cobra.Command, args []string) {
-			createModule(kubeconfig.Moduleset, args[0], repo, path, version, namespace, valuesFile)
+			// Custom validation:
+			// Either templateName or (repo and path) must be provided, if one is provided the other must be empty
+			if (templateName != "" && (repo != "" || path != "")) || (templateName == "" && (repo == "" || path == "")) {
+				log.Fatalf("Error: Either template or (repo and path) must be provided.")
+			}
```
Comment (petar-cvit) [inline cyctl/internal/create/modules.go:93]: Can you also add version here?
```suggestion
			if (templateName != "" && (repo != "" || path != "" || version != "")) || (templateName == "" && (repo == "" || path == "")) {
				log.Fatalf("Error: Either template or (repo, path and version) must be provided.")
			}
```

**Review by petar-cvit — CHANGES_REQUESTED** [review]
Left one comment. Thanks @shivamrazorpay for the PR

File: cyctl/internal/create/modules.go (hunk 1 above)
Comment (shivamrazorpay) [reply cyctl/internal/create/modules.go:93]: Version was not a compulsory flag that is why I left it. I have added it now.

Comment (petar-cvit) [reply cyctl/internal/create/modules.go:93]: Makes sense @shivamrazorpay. This is just users dont use the --version flag to override the version in the template.
Can you push the update?

**Review by petar-cvit — APPROVED** [review]
Thanks @shivamrazorpay 🧡

**Pull request merged**